### ISSUE TITLE
no-dupe-args: Accept a argsIgnorePattern, like no-unused-vars

### ISSUE THREAD

**Issue by mcandre** [eslint/eslint#12149]
**What rule do you want to change?**

no-dupe-args.

**Does this change cause the rule to produce more or fewer warnings?**

Fewer.

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option, based on no-unused-vars' `argsIgnorePattern` option.

**Please provide some example code that this change will affect:**

```js
var baseURI = 'https://localhost:8080/';

// redirectURL accepts an optional request, response HTTP handler pair,
// generating a redirect URI.
function redirectURI(_, _) {
    return baseURI + '/login';
}
```

**What does the rule currently do for this code?**

ESLint parser fails with `error  Parsing error: Argument name clash`. This happens even when the `no-dupe-args` rule is disabled.

**What will the rule do after it's changed?**

Match the behavior of no-unused-vars better, by accepting a similar `"argsIgnorePattern": "^_$"` configuration.

**Are you willing to submit a pull request to implement this change?**

Nope.

**Comment by mdjermanovic** [comment]
Hi @mcandre, thank you for the suggestion.

I personally don't think that ESLint should have a rule/option that would encourage use of a legacy language feature, which is also available only in non-strict mode.

Parser fails most likely because your code is in `strict` mode (explicit or implicit e.g. a module). Duplicate parameter names are forbidden in `strict` mode.

**Comment by mysticatea** [comment]
> ESLint parser fails with error Parsing error: Argument name clash. This happens even when the no-dupe-args rule is disabled.

Because it's syntax error by the language spec. If you are using ES modules or strict mode, you cannot use duplicate names in arguments.

I agree with @mdjermanovic.

The `argsIgnorePattern` option for duplicate arguments is opposing the direction of the language. We would not want to encourage use of that.
